cfme/infrastructure/cluster.py

```diff
@@ -319,19 +299,14 @@ class Details(CFMENavigateStep):
 
     def step(self, *args, **kwargs):
         """Navigate to the correct view"""
-        self.prerequisite_view.toolbar.view_selector.select('List View')
-        version = self.obj.appliance.version
-        if (version >= '5.7.4' and version < '5.8') or version >= '5.8.1.2':
-            cluster_name = self.obj.short_name
-        else:
-            cluster_name = self.obj.name
+        # todo: figure out why the same cfme version shows clusters with short and long name
```

Comment:
**QUESTION:** It randomly chooses a name?

Reply:
clusters appear with either short name f.e. "Default" or long name "Default in Default". 
I thought it depends on MiQ version but I ran into cases when appliances of the same version showed either short names or long names. So, this piece of code tries to use long name if cluster with short name was not found.

Honestly, I think this needs more investigation but I'd like to get around this issue for investigation time.

Reply:
I'm okay with that for now with the TODO.